db/httpdb.js

```diff
@@ -439,6 +440,54 @@ module.exports = function (
     )
   }
 
+  DB.prototype.lockAccount = function (account) {
+    var unlockCode = crypto.randomBytes(16).toString('hex');
+    log.trace({ op: 'DB.lockAccount', uid: account && account.uid, unlockCode: unlockCode })
+
+    return this.pool.post(
+      '/account/' + account.uid.toString('hex') + '/lock',
+      {
+        lockedAt: Date.now(),
+        unlockCode: unlockCode
+      }
+    )
+  }
+
+  DB.prototype.unlockAccount = function (account) {
+    log.trace({ op: 'DB.unlockAccount', uid: account && account.uid })
+    return this.pool.post(
+      '/account/' + account.uid.toString('hex') + '/unlock'
+      )
+      .then(null,
+        function (err) {
+          if (err.statusCode === 404) {
+            // TODO - throw a better error here.
+            err = error.unknownAccount()
+          }
+          throw err
+        }
+      )
+  }
+
+  DB.prototype.unlockCode = function (account) {
+    log.trace({ op: 'DB.unlockCode', uid: account && account.uid })
+    return this.pool.get(
+      '/account/' + account.uid.toString('hex') + '/unlockCode'
+      )
+      .then(
+        function (body) {
+          return body.unlockCode
+        },
+        function (err) {
+          if (err.statusCode === 404) {
+            // TODO - throw a better error here.
+            err = error.unknownAccount()
```

Comment:
I am now throwing the `accountNotLocked` error in the [account/unlock/resend_code route](https://github.com/mozilla/fxa-auth-server/pull/867/files#diff-43da5bcdd83eb1d30aa0b5e9e7473834R609).